flaml/tune/searcher/blendsearch.py

```diff
@@ -314,11 +314,12 @@ def set_search_properties(
                     self.cost_attr = self._ls.cost_attr = TIME_TOTAL_S
             if "metric_target" in spec:
                 self._metric_target = spec.get("metric_target")
-            if "num_samples" in spec:
+            num_samples = spec.get("num_samples")
+            if num_samples is not None:
                 self._num_samples = (
-                    spec["num_samples"]
-                    + len(self._result)
-                    + len(self._trial_proposed_by)
+                    (num_samples + len(self._result) + len(self._trial_proposed_by))
+                    if num_samples > 0

```

Comment:
Just to make sure – what is the expected behavior for `num_samples=0`? Does this cover that case appropriately?

Reply:
`num_samples=0` is treated the same as `num_samples=-1`. This behaviour has not been documented. So I added a comment.